src/Gaufrette/Adapter/BackblazeB2Storage.php

```diff
@@ -0,0 +1,234 @@
+     */
+    public function setBucket($bucket)
+    {
+        $this->bucket = $bucket;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read($key)
+    {
+        $this->ensureBucketExists();
+        try{
+            $result =  $this->service->get($this->bucketName, $this->computePath($key),$this->options["private"]);
+            if(!is_array($result)) return false;
+            return $result["content"];
+        }catch (B2Exception $e){
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write($key, $content)
+    {
+        $this->ensureBucketExists();
+        try {
+            $object = $this->service->insert($this->bucket, $content, $this->computePath($key));
+            if(!is_array($object)) return false;
+            return $object["contentLength"];
+        } catch (B2Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists($key)
+    {
+        $this->ensureBucketExists();
+        try {
+            return $this->service->exists($this->bucketName, $this->computePath($key));
+        } catch (B2Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function keys()
+    {
+        $list = $this->service->all($this->bucket);
+
+        $keys = array();
+        foreach ($list as $file) {
+            $keys[] = $file["fileName"];
+        }
+        sort($keys);
+
+        return $keys;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function mtime($key)
+    {
+        $this->ensureBucketExists();
+
+        $response = $this->service->get(
+            $this->bucketName,
+            $this->computePath($key),
+            $this->options["private"],
+            true
+        );
+
+        return isset($response['X-Bz-Info-src_last_modified_millis']) ? strtotime($response['X-Bz-Info-src_last_modified_millis']) : false;
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        $this->ensureBucketExists();
+
+        try {
+            return $this->service->delete($this->bucketName,  $this->computePath($key), $this->options["private"]);
+        } catch (B2Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function rename($sourceKey, $targetKey)
+    {
+        $this->ensureBucketExists();
+
+        try {
+            return $this->service->rename(
+                $this->bucket,
+                $this->bucketName,
+                $this->computePath($sourceKey),
+                $this->bucketName,
+                $this->computePath($targetKey),
+                $this->options["private"]);
+        } catch (B2Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isDirectory($key)
+    {
+        if ($this->exists($key . '/')) {
```

Comment:
I am not familiar with `Backblaze`, but what happens if I pass `/foo/bar/` directory as parameter here?